New serverless pattern - Add REST API Gateway with VPC Link V2 to Private ALB pattern

agawanea opened the pull request:
Issue https://github.com/aws-samples/serverless-patterns/issues/2869

This pattern demonstrates REST API Gateway integration with a private Application Load Balancer and ECS Fargate cluster using VPC Link V2.

**What's included:**
- REST API Gateway with proxy integration
- VPC Link V2 for private connectivity to ALB
- Private Application Load Balancer
- ECS Fargate cluster with nginx
- Complete security groups and IAM roles
- Terraform implementation

Tested in: us-east-1

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

agawanea committed:
Update author bio in Terraform pattern

agawanea committed:
Merge branch 'aws-samples:main' into add-apigw-rest-vpclink-pattern

agawanea committed:
Update pattern documentation and fix issues

- Update prerequisites to specify NAT Gateway only
- Improve 'How it works' section for better clarity
- Fix testing instructions to include path in curl examples
- Fix LinkedIn format to username only
- Update copyright year to 2025

agawanea committed:
Apply consistent improvements to Terraform pattern

- Use full AWS service names (Amazon API Gateway, Amazon VPC Link V2, Amazon ALB, Amazon ECS)
- Add problem statement explaining VPC Link V2 eliminates NLB requirement
- Update copyright year to 2026

biswanathmukherjee commented:
Hi @agawanea, Are you updating the pattern based on the comments provided?

agawanea committed:
Merge branch 'aws-samples:main' into add-apigw-rest-vpclink-pattern

agawanea committed:
fix: Address PR review comments

- Remove project-level .gitignore (repo-level exists)
- Fix title to 'Amazon API Gateway REST API to Private ALB via VPC Link V2'
- Change language from HCL to YAML (valid enum value)
- Remove projectFolder prefix from templateFile
- Restrict ALB security group to VPC CIDR instead of 0.0.0.0/0
- Change 'Update' to 'Create' for terraform.tfvars instructions
- Add architecture diagram

biswanathmukherjee commented on apigw-rest-vpclink-pvt-alb-terraform/README.md:
````diff
@@ -0,0 +1,109 @@
+```bash
+curl https://<API-ENDPOINT>/index.html
+```
+
````
curl https://..../prod/index.html ; echo

I am getting the following while testing:

```
<html>
<head><title>503 Service Temporarily Unavailable</title></head>
<body>
<center><h1>503 Service Temporarily Unavailable</h1></center>
</body>
</html>
```

biswanathmukherjee commented on apigw-rest-vpclink-pvt-alb-terraform/README.md:
```diff
@@ -0,0 +1,109 @@
+* [Create an AWS account](https://portal.aws.amazon.com/gp/aws/developer/registration/index.html) if you do not already have one and log in. The IAM user that you use must have sufficient permissions to make necessary AWS service calls and manage AWS resources.
+* [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html) installed and configured
+* [Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git) installed and configured
+* [Terraform](https://learn.hashicorp.com/tutorials/terraform/install-cli) installed
```
Please mention the min version required. I had an older version of terraform and it threw error.

biswanathmukherjee committed:
Create apigw-rest-vpclink-pvt-alb-terraform.json

biswanathmukherjee commented:
Hi @agawanea , Thanks for the update. However, I am getting runtime issues now. I have added a couple of comments. Can you please update the PR?

agawanea committed:
fix: Address runtime issues from PR review

- Add dedicated VPC Link V2 security group with egress to ALB on port 80
- Attach the SG to VPC Link (was empty, causing network isolation)
- Replace ALB inline ingress with explicit rule from VPC Link SG only
- Add explicit target group health check (path, interval, thresholds)
- Add wait_for_steady_state to ECS service
- Increase health_check_grace_period_seconds from 60 to 120
- Depend VPC Link integration on ECS service being stable
- Document minimum Terraform version (1.9+) in README
- Clarify testing URL includes /prod stage with concrete example
- Add troubleshooting note about transient 503 during rollout

biswanathmukherjee commented:
Thanks @agawanea for your contribution. This is pattern guardian approved. DA will review/ merge soon.
